test/integ/user_login_integ_test.js

```diff
@@ -0,0 +1,122 @@
+const databaseOps = require("../../lib/database-ops");
+const usernameUtil = require("../../lib/util/username");
+const server = require("../../lib/server");
+const { MongoMemoryServer } = require('mongodb-memory-server');
+const { MongoClient } = require("mongodb");
+const { assert } = require("chai");
+const { stub } = require("sinon");
+const { getServerAgent, addUser } = require("./integ_test_utils");
+
+// TODO:#119 shut down mongo mem server and remove --exit hopefully
+
+const performUserLogin = (agent, username, password) => {
+    return agent.post("/login")
+            .type("form")
+            .send({
+                username,
+                password,
+            });
+};
+
+describe("integ", () => {
+    describe("user login", () => {
+        var mongod = null;
+        var db = null;
+        var agent = null;
+        var usersCollection = null;
+        const TEST_USER = "test-user";
+        const TEST_PASSWORD = "test-password";
+
+        it("should be able to login user successfully", () => {
+            return performUserLogin(agent, TEST_USER, TEST_PASSWORD)
+                .then(res => {
+                    assert.equal(res.status, 200);
+                    assert.equal(res.body.username, TEST_USER);
+                });
+        });
+
+        it("should not be able to login user if username is incorrect", () => {
+            return addUser("some-user", "some-password", "incorrect-user@test.com")
+                .then(() => {
+                    return performUserLogin(agent, "wrong-user", "some-password");
+                })
+                .then(res => {
+                    assert.equal(res.status, 401);
+                    assert.equal(res.body.errorID, "userPassComboNotFound");
```

Comment:
For each error login case, add the following:
- Pre-condition: /user endpoint returns error for no active session
- Post-condition: /user endpoint returns error for no active session

In the future, might be a good idea to have a test util that interfaces directly with redis to get the active session instead of rely on /user endpoint.